commands/skills/appraise.py

```diff
@@ -0,0 +1,142 @@
+        
+        if not hasattr(caller.ndb, "appr_win") or not caller.ndb.appr_win:
+            caller.ndb.appr_win = {}
+
+        # check if the object has failed to be appraised recently. If so,
+        # prohibit further appraisal.
+
+        #caller.msg("initial lose list: " + str(caller.ndb.appr_lose))
+        #caller.msg("initial win list: " + str(caller.ndb.appr_win))
+
+        lose = caller.ndb.appr_lose
+        reroll_time = 3600 # how much time one has to wait before rerolling
+                           # the appraise skill on the same type of item
+
+        if lose.has_key(obj.name):
+            if ct - lose[obj.name] <= reroll_time:
+                #caller.msg("Using lose record from %f" % lose[obj.name])
+
+                caller.msg("You have already attempted to appraise %s " %
+                    obj.name +
+                    "recently and must wait a while before trying again.")
+                return
+            else:
+                # clear the record of the item
+                #caller.msg("Clearing appraise check loss record from %f" %
+                #    lose[obj.name])
+
+                lose.pop(obj.name)        
+
+        # check if the object has already been successfully appraised recently.
+        # if not, run the appraise skill check.
+
+        win = caller.ndb.appr_win
+
+        if win.has_key(obj.name):
+            if ct - win[obj.name] <= reroll_time:
+                #caller.msg("Using win record from %f" % win[obj.name])
+
+                self.display(obj)
+                return
+            else:
+                    # clear the record of the item
+                #caller.msg("Clearing appraise check win record from %f" %
+                #    win[obj.name])
+
+                caller.ndb.appr_win.pop(obj.name)
+
+            # run the skill check again
+        if skill_check(caller.skills.appraise.actual):
+            #caller.msg("Appraise skill check successful")
+            self.display(obj)
+
+               #add a record of the item to the win list
+            win[obj.name] = ct
+        else:
+            #caller.msg("Appraise skill check unsuccessful")
+
+            caller.msg("You cannot tell the value of %s." % obj.name)
+
+               #add a record of the item to the lose list
+            lose[obj.name] = ct
+
+
+        #caller.msg("final lose list: " + str(lose))
+        #caller.msg("final win list: " + str(win))
+
+        return
+
+    def display(self, obj):
+        # display the value of the item
+        coin = value_to_coin(obj.db.value)
+
+        if coin:
+            cc = "%d CC" % coin['CC']
+            if coin['GC'] == 0:
+                gc = ""
+                sc = " and "
+            else:
+                gc = " and %d GC" % coin['GC']
+                sc = ", "
+            if coin['SC'] == 0:
+                sc = ""
+            else:
+                sc += "%d SC" % coin['SC']
+            self.caller.msg("The value of %s is %s%s%s" % 
```

Comment:
For consistency, would you be willing to use the world.economy.format_coin function here?
Also, we've been trying to move to using {} style text format syntax and away from %s. Could that happen here too?